### PR TITLE
bandaid mparticle v3, update snapchat icon

### PR DESCRIPTION
**Pull request by johndavidsimmons**
Mparticle has released version 3 of their web SDK which is fundamentally different from v1 & v2. I have put a few patches in place to make sure the calls show up in Omnibug, but it will need a complete refactor, if not a separate provider class entirely to classify the parameters to match v1/v2 calls.

Also, the snap chat icon was white on white. I changed it to use the white on yellow icon which is what they use in their favicon.

### CODE REVIEW

File: src/providers/Mparticle.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ class MparticleProvider extends BaseProvider {
     constructor() {
         super();
         this._key       = "MPARTICLE";
-        this._pattern   = /\.mparticle\.com\/v\d\/JS\/[^/]+\/Events$/;
+        this._pattern   = /\.mparticle\.com\/v\d\/JS\/[^/]+\/events/i;
```
Comment (MisterPhilip) [inline src/providers/Mparticle.js:13]: Hmm, we may need to rethink this part - either changing it to `[eE]` or having _all_ of the patterns be case-insensitive.

per https://github.com/MisterPhilip/omnibug/blob/master/src/providers/OmnibugProvider.js#L39, we have:
```js
let x = new RegExp(defaultPattern.map((el) => {
                return el.source;
            }).join("|"));
```

So once we run that in context of multiple providers:
```js
let defaultPattern = [/tESt/i, /fooBAR/];
let x = new RegExp(defaultPattern.map((el) => {
                return el.source;
            }).join("|"));
// x is now /tESt|fooBAR/ - note the missing case insensitive flag for tESt

x.test("test"); // returns false
```

**Review by MisterPhilip — CHANGES_REQUESTED** [review]
The easiest fix here is changing `events` to `[eE]vents`. I'm a little worried of making all provider patterns case-insensitive, but let me know if you have thoughts on it!

File: src/providers/Mparticle.js (hunk 1 above)
Comment (johndavidsimmons) [reply src/providers/Mparticle.js:13]: I think changing it to just [eE] is ok for now. Then revisit the bigger conversation when doing a real provider upgrade for v3. I will make this change and resubmit

Comment (MisterPhilip) [reply src/providers/Mparticle.js:13]: No need for a full resubmit, just commit a new change and it'll add it on to the pull request 😄

**Review by MisterPhilip — APPROVED** [review]

File: src/providers/Mparticle.js (hunk 1 above)
Comment (johndavidsimmons) [reply src/providers/Mparticle.js]: done

**Comment by MisterPhilip** [comment]
Awesome, thanks for this! Let me know what you want to do/talk through for v3; I'm currently in the same boat for Target at.js v2 trying to figure out if it's better for a separate provider or merge it all into one.

**Pull request merged**